src/compiler/checker.ts

```diff
@@ -4738,6 +4739,14 @@ namespace ts {
             return checkTypeAssignableTo(source, target, /*errorNode*/ undefined);
         }
 
+        /**
+         * This is *not* a bi-directional relationship.
+         * If one needs to check both directions for comparability, use a second call to this function or 'checkTypeComparableTo'.

```

Comment:
All uses *do* need to check both directions, so probably you should have a version that does that. You might not even need the uni-directional ones except for consistency with the other relations.